app/http.php

```diff
@@ -764,6 +769,8 @@ function (string $runtimeId, string $payload, array $variables, int $timeout, st
             $runtime['updated'] = \microtime(true);
             $activeRuntimes->set($activeRuntimeId, $runtime);
 
+            $activeRuntimes->decr($activeRuntimeId, 'executions', 1);
```

Comment:
The get and set from the lines above this make me worried because those can introduce a race condition 😕

Reply:
I did some testing and it seems like the `set()` method supports partial update. Here's my code:

```php
<?php

require_once __DIR__ . '/vendor/autoload.php';

use Swoole\Table;

$table = new Table(1024);

$table->column('name', Table::TYPE_STRING, 256);
$table->column('created', Table::TYPE_FLOAT);
$table->column('updated', Table::TYPE_FLOAT);
$table->column('status', Table::TYPE_STRING, 128);
$table->create();

$startTime = \microtime(true);

$result = $table->set('key', [
    'name' => 'coolname',
    'created' => $startTime,
    'updated' => $startTime,
    'status' => 'pending',
]);

$row = $table->get('key');

var_dump($row);

$result = $table->set('key', [
    'updated' => \microtime(true),
    'status' => 'started',
]);

$row = $table->get('key');

var_dump($row);
```

And here's the output:

```
array(4) {
  ["name"]=>
  string(8) "coolname"
  ["created"]=>
  float(1679511513.4115)
  ["updated"]=>
  float(1679511513.4115)
  ["status"]=>
  string(7) "pending"
}
array(4) {
  ["name"]=>
  string(8) "coolname"
  ["created"]=>
  float(1679511513.4115)
  ["updated"]=>
  float(1679511513.411552)
  ["status"]=>
  string(7) "started"
}
```

As you can see, I was able to update `updated` and `status` without modifying `name` and `created`. If we make sure not to include `executions` in the `set()`, I think we can avoid the race condition.